### PR TITLE
Add ultralight package

### PR DESCRIPTION
**Pull request by ArthurVasseur**
This PR adds the `ultralight` package

### CODE REVIEW

File: packages/u/ultralight/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+         assert(package:check_cxxsnippets({test = [[
+            #include <AppCore/App.h>
+            #include <Ultralight/platform/Platform.h>
+
+            void test()
+            {
+                auto app = ultralight::App::Create();
+            }
+        ]]}, {configs = {languages = "c++17"}}))
+
```
Comment (waruqi) [inline packages/u/ultralight/xmake.lua:30]: `end)` is missing

File: packages/u/ultralight/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+package_end()
```
Comment (waruqi) [inline packages/u/ultralight/xmake.lua:31]: please remove it

File: packages/u/ultralight/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package("ultralight")
+    set_homepage("https://ultralig.ht")
+    set_description("Ultralight makes it easy for C/C++ developers to seamlessly integrate web-content into games and desktop apps.")
+    set_license("LGPL")
+
+    if (is_plat("windows")) then
```
Comment (waruqi) [inline packages/u/ultralight/xmake.lua:6]: remove `()`, `if is_plat("windows") then`

File: packages/u/ultralight/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    else
+        add_urls("https://github.com/ultralight-ux/Ultralight/releases/download/v$(version)/ultralight-sdk-$(version)-$(plat)-$(arch).7z")
+    end
+
+    add_versions("1.3.0", "cc8bfc66a4c40c88fa02691febe6f21c248a2a30d17cfe5470fccc3a461ce49e")
+
+    on_install("windows", "linux", "macosx", function (package)
+        os.cp("include", package:installdir())
+        os.cp("lib", package:installdir())
+        os.cp("bin", package:installdir())
```
Comment (waruqi) [inline packages/u/ultralight/xmake.lua:17]: I don't think the linux binary is compatible with all linux distributions and compilers.

Comment (ArthurVasseur) [reply packages/u/ultralight/xmake.lua:17]: Yes, according to their website 
> Ultralight supports most major desktop PC platforms. Specifically, Windows (7+, x64), macOS (Sierra or later), and Linux (Ubuntu/Debian 9.5+, x64).

I will fix that in my next commit

File: packages/u/ultralight/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    on_install("windows|x64", "linux|x86_64", "macosx|x86_64", function (package)
+        if package:is_plat("linux") then
+            if linuxos.name() ~= "ubuntu" and linuxos.name() ~= "debian" or (linuxos.version():major() < 9 and linuxos.version():minor() < 5) then
+                print("Ultralight is officially supported on Ubuntu/Debian 9.5+. use it at your own risks")
+            end
+        end
+        os.cp("include", package:installdir())
+        if not package:is_plat("macosx") then
+            os.cp("lib", package:installdir())
```
Comment (waruqi) [inline packages/u/ultralight/xmake.lua:27]: no lib directory in linux tar.gz

**Pull request merged**